docs/api.md

```diff
@@ -73,3 +73,80 @@ function example_civicrm_mosaicoConfig(&$config) {
   $config['tinymceConfigFull']['toolbar1'] = 'bold italic removeformat | link unlink | civicrmtoken | pastetext code';
 }
 ```
+
+## hook_civicrm_mosaicoStyleUrlsAlter(&$styleUrls)
+
+`$styleUrls` is an array of URLs for stylesheets to be loaded in the Mosaico composer interface. Alter (add, remove, change) array elements as needed.
+
+Example from  [Campaign Advocacy](https://github.com/twomice/civicrm-campaignadvocacy):
+
+```php
+function campaignadv_civicrm_mosaicoStyleUrlsAlter(&$styleUrls) {
+  $res = CRM_Core_Resources::singleton();
+  $snippets = array_merge(Civi::service('bundle.coreResources')->getAll(), Civi::service('bundle.coreStyles')->getAll());
+
+  // crm-i.css added ahead of other styles so it can be overridden by FA.
+  array_unshift($styleUrls, $res->getUrl('civicrm', 'css/crm-i.css', TRUE));
+
+  foreach ($snippets as $snippet) {
+    $itemUrl = NULL;
+    if (
+      FALSE !== strpos($snippet['name'], 'css')
+      // Exclude jquery ui theme styles, which conflict with Mosaico styles.
+      && FALSE === strpos($snippet['name'], '/jquery-ui/themes/')
+    ) {
+      if ($snippet['styleUrl'] ?? FALSE) {
+        $itemUrl = $snippet['styleUrl'];
+      }
+      elseif ($snippet['styleFile'] && count($snippet['styleFile']) == 2) {
+        $itemUrl = $res->getUrl($snippet['styleFile'][0], $snippet['styleFile'][1], TRUE);
+      }
+      if ($itemUrl) {
+        $styleUrls[] = $itemUrl;
+      }
+    }
+  }
+
+  // Include our own abridged styles from jquery-ui 'smoothness' theme, as
+  // required for our jquery-ui dialog, but which don't conflict with Mosaico.
+  $styleUrls[] = $res->getUrl('campaignadv', 'css/jquery-ui-smoothness-partial.css', TRUE);
+}
+```
+
+## hook_civicrm_mosaicoScriptUrlsAlter(&$scriptUrls)
+
+`$scriptUrls` is an array of URLs for JavaScript files to be loaded in the Mosaico composer interface. Alter (add, remove, change) array elements as needed.
+
+Example from  [Campaign Advocacy](https://github.com/twomice/civicrm-campaignadvocacy):
+
+```php
+function campaignadv_civicrm_mosaicoScriptUrlsAlter(&$scriptUrls) {
+  $res = CRM_Core_Resources::singleton();
```

Comment:
@lucky091588 I recommend removing lines 124-147 because they'll be confusing for people who just want to know how this hook works. I would say that lines 148-150 are enough for this example.